docs/tutorial/new-tutorial.md

```diff
@@ -0,0 +1,384 @@
+(tutorial)=
+# SQLair tutorial
+
+Imagine you are writing a Go program that handles your customers and their
+orders. The business logic is contained in the Go program but the customer and
+order data is stored in a SQL database. In this tutorial, we'll see how easy
+SQLair makes it to map the database between the database and the Go program.
+Let's get started!
+
+
+```{note}
+This tutorial will take around 30 minutes to complete. If you want a quicker
+start see {ref}`prepare-and-run`.
+```
+
+## Setting up the environment
+
+To start with create a new folder called tutorial and initialise a new Go project.
+```bash
+mkdir tutorial 
+cd tutorial
+go mod init sqlair-tutorial
+```
+
+We first need to set up the backbone of our project, the database. The easiest
+way to create a SQL database in Go is to use [SQLite](https://www.sqlite.org/)
+with the [Go driver from mattn](https://github.com/mattn/go-sqlite3) though
+SQLair works with any database driver that can interface with
+[database/sql](https://pkg.go.dev/database/sql). With SQLite, we can create an in
+process, in memory database that can be populated within the program and that
+will disappear when the program finishes running.
+
+We also want to set the database up with our customer and order schema. We will
+have two tables. One for customers and the other for orders.
+
+To create the database, create a new `main.go` file in the project, open it in
+your favourite editor and copy in the code below.
+```go
+package main
+
+import (
+    "database/sql"
+    "fmt"
+
+    "github.com/canonical/sqlair"
+    _ "github.com/mattn/go-sqlite3"
+)
+
+// NewDB opens and returns a new SQLair database.
+func NewDB() (*sqlair.DB, error) {
+    sqlDB, err := sql.Open("sqlite3", "file:sqlair-tutorial.db?cache=shared&mode=memory")
+    if err != nil {
+        return nil, fmt.Errorf("opening SQLite database: %w", err)
+    }
+
+    _, err = sqlDB.Exec(`
+    CREATE TABLE customers (
+        id integer PRIMARY KEY,
+        name text,
+        address text,
+        city text, 
+        postal_code text,
+        country text
+    );
+
+    CREATE TABLE orders (
+        id integer PRIMARY KEY,
+        customer_id integer,
+        order_date text,
+        FOREIGN KEY (customer_id) REFERENCES customer (id)
+    );
+`)
+    if err != nil {
+        return nil, fmt.Errorf("creating tables: %w", err)
+    }
+
+    db := sqlair.NewDB(sqlDB)
+    return db, nil
+}
+
+func tutorial() error {
+    _, err := NewDB()
+    if err != nil {
+        return err
+    }
+    return nil
+}
+
+func main() {
+    err := tutorial()
+    if err != nil {
+        fmt.Printf("ERROR: %s\n", err)
+    }
+    fmt.Println("Finished without errors!")
+}
+```
+The program above will open the SQLite database, add our schema to it, then wrap
+it as a SQLair DB. SQLair does not create the underlying database, only provide
+an interface for getting and setting the values within it.
```

Comment:
Avoid use both "below" and "above" -- choosing a single direction usually makes for better flow. Here I'd just say: This will open the...